cypress/e2e/immutableDatabaseTests/linkCheck.ts

```diff
@@ -0,0 +1,104 @@
+import { setTokenUserViewPort } from '../../support/commands';
+
+function formatPaths(paths: string[]) {
+  let formattedPaths = '';
+  paths.forEach((path) => {
+    formattedPaths += '\n- ' + path;
+  });
+  return formattedPaths;
+}
+
+describe('Find broken anchor links', () => {
+  setTokenUserViewPort();
+
+  function checkUrls(path: string, selector: string) {
+    let brokenUrls = [];
+    cy.visit(path);
+    cy.get(selector)
+      .find('a')
+      .each((url) => {
+        cy.get(url).then((url) => {
+          const href = url.prop('href');
+          if (href) {
+            cy.request({
+              url: href,
+              failOnStatusCode: false,
+            }).then((result) => {
+              if (result.status != 200) {
+                brokenUrls.push(href);
+                cy.log(result.status + ': ' + href);
+              }
+            });
+          }
+        });
+      })
+      .then(() => {
+        if (brokenUrls.length) {
+          throw new Error(`Broken links at "${path}":` + formatPaths(brokenUrls));
+        }
+      });
+  }
+
+  const urlPages = [
+    ['/', 'app-root'],
+    ['/sitemap', 'app-sitemap'],
+    ['/about', 'app-about'],
+    ['/funding', 'app-funding'],
+    ['/docs', 'app-docs'],
+    ['/dashboard', 'app-my-sidebar'],
+  ];
+
+  urlPages.forEach((urlPage) => {
+    it(`anchor links at "${urlPage[0]}" should work`, () => {
+      checkUrls(urlPage[0], urlPage[1]);
+    });
+  });
+});
+
+describe('Find broken image links', () => {
+  setTokenUserViewPort();
+
+  function checkImages(path: string, selector: string) {
+    let brokenImages = [];
+    cy.visit(path).wait(500);
+    cy.get(selector)
+      .find('img')
+      .each((image) => {
+        cy.get(image).then((image) => {
+          if (image.prop('naturalWidth') === 0) {
+            brokenImages.push(image.prop('src'));
+            cy.log('image not visible: ' + image.prop('src'));
+          }
+        });
+      })
+      .then(() => {
+        if (brokenImages.length) {
+          throw new Error(`Broken images at "${path}":` + formatPaths(brokenImages));
+        }
+      });
+  }
+
+  const imagePages = [
+    ['/', 'app-root'],
```

Comment:
Why do you have the selector, `app-root`? Is it to avoid checking the same image twice? E.g., if you go to `/about`, it's a child of `app-about` is a child of `app-root`, so you don't want to check `app-root` again?

Another idea is you could have a set of visited images, so that you don't visit the same image twice.

It would make it easier to maintain -- if we add a new path, e.g., `/wonderful`, we wouldn't have to know the component name to add it to this array.

Reply:
I think using `app-root` also conveniently performs link checks for `app-navbar` and `app-footer`, alongside the main home component `app-home`. We could also replace it with `['/', 'app-navbar'], ['/', 'app-footer'], ['/', 'app-home']` to make it clearer.

Reply:
Not sure if this is what you meant by a set of visited images: the test could dynamically add images to a list after they've been visited, so they can be skipped if they appear a second time? Would that significantly add to the test's runtime?

Another thought: if each path had a corresponding list of images to check, then the test could visit each by their `src` tag? This way, we could write the test to avoid duplicate checks: if an image exists on 2+ pages, it only has to be visited on one of them.

^^ Pros: This might also allow us to better filter by which images should (or shouldn't) be checked. For instance, when visiting /organizations, we might not be responsible for an organization's avatar image, but we do want to verify that the organization icon is visible.

^^ Cons: While it does remove the need for component names, you'd have to add a `src` path for every new image, so I can't tell if that makes it easier or harder to maintain. It also might be a bit of effort upfront to cumulate all the images.